packages/manager/modules/iam/src/components/actionSelect/ActionTrees.class.js

```diff
@@ -363,13 +364,15 @@ export default class ActionTrees extends Array {
           this.buffer.actions.find(({ value }) => value === action),
         )
         .filter(Boolean);
+      const currentActions = actions.map((action) => ({
+        ...action,
+        id: `${actionTree.value}_${category.value}_${action.value}`
+          .replace(/[^\w]/g, '-')
+          .toLocaleLowerCase(),
+      }));
       Object.assign(category, {
-        actions: actions.map((action) => ({
-          ...action,
-          id: `${actionTree.value}_${category.value}_${action.value}`
-            .replace(/[^\w]/g, '-')
-            .toLocaleLowerCase(),
-        })),
+        actions: currentActions,
+        filteredActions: currentActions,
```

Comment:
**nitpick:** Why maintain the same array twice in the same object?

Reply:
Hello, this is just initialization of the value. Whenever a user uses a filter, it is updated. I need to have an intermediate value because I need a reference with all actions so I can reset to original state actions if user deletes the filter, or updates it. Seems ok to me.